eng/common/pipelines/ci.yml

```diff
@@ -0,0 +1,107 @@
+trigger: none
+pr:
+  branches:
+    include:
+      - checkEnforcer
+
+extends:
+  template: /eng/common/pipelines/templates/1es-redirect.yml
+  parameters:
+    stages:
+      - stage: InitStage
+        displayName: Initialize
+        pool:
+          name: $(LINUXPOOL)
+          image: $(LINUXVMIMAGE)
+          os: linux
+        jobs:
+          - job: InitJob
+            displayName: Initialize
+            steps:
+              - pwsh: |
+                  Write-Host "Setting RunCore to true"
+                  Write-Host "##vso[task.setvariable variable=RunCore;isOutput=true]true"
+                  Write-Host "Setting RunCSharp to true"
+                  Write-Host "##vso[task.setvariable variable=RunCSharp;isOutput=true]true"
+                  Write-Host "Setting RunPython to true"
+                  Write-Host "##vso[task.setvariable variable=RunPython;isOutput=true]true"
+                name: InitStep
+                displayName: Initialize
+
+      # Run csharp stages if RunCSharp == true
+      - template: /packages/http-client-csharp/eng/pipeline/templates/ci-stages.yml
+        parameters:
+          DependsOn: InitStage
+          Condition: eq('true', stageDependencies.InitStage.outputs['InitJob.InitStep.RunCSharp'])
+
+      # Run core stages if RunCore == true
+      - stage: Build_Core
+        displayName: Build Core
```

Comment:
nit but could we just name the stage to be the product so we reduce the text on the github status list

Reply:
It allows for multiple stages for a given product like this
![image](https://github.com/microsoft/typespec/assets/64171366/c9ee6e14-08b6-49d2-81cb-04193ac96652)

Reply:
hhm i see guess its ok then

Reply:
We could reduce each product to a single stage, so there would just be:
```mermaid
flowchart LR
    Initialize --> Core & CSharp & Python
```

Reply:
Stages are just logical groupings for jobs. We should be able to combine them easily